x-pack/plugins/enterprise_search/server/lib/indices/pipelines/ml_inference/pipeline_processors/delete_ml_inference_pipeline.ts

```diff
@@ -18,6 +21,12 @@ export const deleteMlInferencePipeline = async (
 ) => {
   let response: DeleteMlInferencePipelineResponse = {};
 
+  // Check if the pipeline is in use in a different index's managed pipeline
+  const pipelineIsInUse = await isPipelineInUse(pipelineName, indexName, client);
+  if (pipelineIsInUse) {
+    throw new Error(ErrorCode.PIPELINE_IS_IN_USE);
```

Comment:
It would be useful if we returned the name of the other index here, embedded in the thrown `Error` object. However, the `Error` constructor doesn't accept arbitrary attributes. Is there any other type we can use here? (I also looked at `EnterpriseSearchError` but it doesn't support attributes either.)

Reply:
The argument to `Error` is just a string, right? Idk how you feel about this, but we could interpolate `ErrorCode.PIPELINE_IS_IN_USE` and the error message (including the name of the other pipeline) and then a possible implementation of `isPipelineIsInUseException` could check for `ErrorCode.PIPELINE_IS_IN_USE` within the string instead of an equality check... 🤷

Reply:
@sphilipse The challenge is that the conflicting index name will be something other than `indexName`; in `isPipelineInUse()` we're checking if the pipeline is referenced somewhere else than in `indexName`'s managed pipeline.

So we'd need two changes:
1. Make `isPipelineInUse()` return `[boolean, string]` or some other construct that contains the problematic index name.
2. Add that property to whatever we're throwing here.

For the latter I was thinking introducing a new class, e.g.
```
class EnterpriseSearchInternalError extends Error {
  data: Record<string, any> | undefined;

  constructor(message?: string, data?: Record<string, any>) {
    super(message);
    this.data = data;

    // We are extending a built-in class
    Object.setPrototypeOf(this, EntSearchError.prototype);
  }
}

// If pipeline is in use:
throw new EnterpriseSearchInternalError('...', { otherIndexName: ... });
```

However I'm not sure if this is the right pattern / approach.

Reply:
Done, I ended up returning `pipelineName` alongside the message.

Reply:
> you don't actually need to throw a class. You can throw any kind of object. So just throwing:

Apparently the linter isn't happy about this: https://buildkite.com/elastic/kibana-pull-request/builds/83336#01841afc-ca49-410b-bdf3-6a999c6e490d
```
25:5  error  Expected an error object to be thrown  no-throw-literal
```

Could you suggest another way that both satisfies the linter and supports additional properties? E.g.
```
    throw Object.assign(new Error(ErrorCode.PIPELINE_IS_IN_USE), {
      pipelineName: otherPipelineName
    });
```
?

Reply:
I guess your suggestion would work, I'd be inclined to just override the linter. That linting rule exists because some browsers don't deal well with non-Error object errors, but that's not really a problem because we're running in Node.

Reply:
You could construct your own Error class the way you wanted to initially, but I'd probably restrict to just this usage and not make it very generic--errors tend not to be very generalizable across use cases.

Reply:
I went with `Object.assign()`, let me know if you want it changed.